### PR TITLE
feat: load modules from multiple paths, support symlink as module

### PR DESCRIPTION
**Pull request by diafour**
#### Overview

- Resolve symlinks in modules directory, use symlink name to determine module name and order
- Support multiple paths to search for modules.
- Support directories without number prefix as module dirs. Default order is 1.
- Remove '3 digits prefix' restriction.
- Load values.yaml from all paths.

#### What this PR does / why we need it

Related issues: 
- https://github.com/flant/addon-operator/issues/262
- https://github.com/deckhouse/deckhouse/pull/3629

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/module_manager/module_manager.go [hunk 1]
```diff
@@ -709,17 +705,15 @@ func (mm *moduleManager) RefreshEnabledState(logLabels map[string]string) (*Modu
...
-	module, exist := mm.allModulesByName[name]
-	if exist {
-		return module
-	} else {
-		log.Errorf("Possible bug!!! GetModule: no module '%s' in ModuleManager indexes", name)
-		return nil
+	if mm.modules.Has(name) {
+		return mm.modules.Get(name)
 	}
+	log.Errorf("Possible bug!!! GetModule: no module '%s' in ModuleManager indexes", name)
```
Comment (shvgn) [inline pkg/module_manager/module_manager.go:711]: It's not a bug, just no such module. It is fine, CALM DOWN AND DON'T PANIC!!!

```
deckhouse-controller module values my-inexisting-module
```

File: pkg/module_manager/module_loader.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+			if !modules.Has(module.Name) {
+				modules.Add(module)
+			}
+		}
+	}
+
+	return modules, nil
+}
+
+const PathsSeparator = ":"
```
Comment (shvgn) [inline pkg/module_manager/module_loader.go:36]: So, PathSeletor is a dedicated single-use constant. While "values.yaml" is just a magic string!

File: pkg/module_manager/module_loader.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+	return res
+}
+
+func findModulesInDir(modulesDir string) ([]*Module, error) {
+	dirEntries, err := os.ReadDir(modulesDir)
+	if err != nil && os.IsNotExist(err) {
+		return nil, fmt.Errorf("path '%s' not exists", modulesDir)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("list modules directory '%s': %s", modulesDir, err)
```
Comment (shvgn) [inline pkg/module_manager/module_loader.go:56]:
```suggestion
		return nil, fmt.Errorf("listing modules directory '%s': %s", modulesDir, err)
```

File: pkg/module_manager/module.go [hunk 4]
```diff
@@ -885,61 +886,30 @@ func (m *Module) runEnabledScript(precedingEnabledModules []string, logLabels ma
...
-// RegisterModules load all available modules from modules directory
-// FIXME: Only 000-name modules are loaded, allow non-prefixed modules.
-func (mm *moduleManager) RegisterModules() error {
-	log.Debug("Search and register modules")
+	mm.commonStaticValues = commonStaticValues
 
 	modules, err := SearchModules(mm.ModulesDir)
 	if err != nil {
 		return err
 	}
```
Comment (shvgn) [inline pkg/module_manager/module.go:908]: So, if name != restoredName, we'll get an error. Is that an intended case?

File: pkg/module_manager/module_manager.go (hunk 1 above)
Comment (diafour) [reply pkg/module_manager/module_manager.go:711]: Good point, yeah. The caller should check for nil.

File: pkg/module_manager/module_loader.go [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+	targetPath, err := resolveSymlinkToDir(dirPath, entry)
+	if err != nil {
+		return "", "", fmt.Errorf("resolve '%s' as a possible symlink: %v", absPath, err)
+	}
+	if targetPath != "" {
+		return name, targetPath, nil
+	}
+
+	if name != "values.yaml" {
+		log.Warnf("Ignore '%s' while searching for modules", absPath)
```
[the remark this answers is not in the document]
Comment (diafour) [reply pkg/module_manager/module_loader.go:98]: Those warnings and errors are cheap substitutions for dry-run/lint mode and ignore file :(

File: pkg/module_manager/module_loader.go (hunk 2 above)
Comment (diafour) [reply pkg/module_manager/module_loader.go:36]: Good point, I'll extract it to the constant.

File: pkg/module_manager/module.go (hunk 4 above)
Comment (diafour) [reply pkg/module_manager/module.go:908]: Yes, kebab-case to camelCase is a conversion to get a values key and camelCase to kebab-case is a conversion when ConfigMap is loaded.

File: RUNNING.md [hunk 6]
```diff
@@ -2,13 +2,15 @@
 
 ## Environment variables
 
-**MODULES_DIR** — a directory where modules are located.
-
 **GLOBAL_HOOKS_DIR** — a directory with global hook files.
 
-**ADDON_OPERATOR_NAMESPACE** — a required parameter with namespace where Addon-operator is deployed.
+**MODULES_DIR** — paths separated by colon where modules are located.
```
[the remark this answers is not in the document]
Comment (diafour) [reply RUNNING.md:7]: I also thought about `MODULES_PATH`

**Review by yalosev — APPROVED** [review]

File: pkg/module_manager/module_manager.go (hunk 1 above)
Comment (shvgn) [reply pkg/module_manager/module_manager.go:711]: it's more about the exclamation ""Possible bug!!!" :)

File: RUNNING.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (shvgn) [reply RUNNING.md:7]: MODULES_PATH**S** !

File: pkg/module_manager/module_manager.go (hunk 1 above)
Comment (diafour) [reply pkg/module_manager/module_manager.go]: Done

File: RUNNING.md [hunk 7]
```diff
@@ -2,13 +2,15 @@
...
+
+**UNNUMBERED_MODULE_ORDER** — a default order for modules without numbered prefix.
```
[the remark this answers is not in the document]
Comment (diafour) [reply RUNNING.md]: Explained

File: RUNNING.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (diafour) [reply RUNNING.md:7]: This rename is a huge thing, it'll require changes in manifests, I don't want to make it in this PR.

**Pull request merged**